### PR TITLE
docs(developer-guide): added phase 1 of dev guide

### PR DESCRIPTION
**Pull request by toniblyx**
### Context

As a developer that wants to contribute I need information on how to stat developing custom checks and frameworks in Prowler

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/tutorials/developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+First of all, you need a version of Python 3.9 or higher and also pip installed to be able to install all dependencies requred. Once that is satisfied go a head and clone the repo:
+
+```
+git clone https://github.com/prowler-cloud/prowler
+cd prowler
+```
+For isolation and avoid conflicts with other environments, we recommend usage of `pipenv`:
+```
+pip install pipenv
```
Comment (jfagoagas) [inline docs/tutorials/developer-guide.md:15]: Change this to `poetry`

File: docs/tutorials/developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+## Get the code and install all dependencies
+
+First of all, you need a version of Python 3.9 or higher and also pip installed to be able to install all dependencies requred. Once that is satisfied go a head and clone the repo:
+
+```
+git clone https://github.com/prowler-cloud/prowler
+cd prowler
+```
+For isolation and avoid conflicts with other environments, we recommend usage of `pipenv`:
```
Comment (MrCloudSec) [inline docs/tutorials/developer-guide.md]:
```suggestion
For isolation and avoid conflicts with other environments, we recommend usage of `poetry`:
```

File: docs/tutorials/developer-guide.md (hunk 1 above)
Comment (MrCloudSec) [inline docs/tutorials/developer-guide.md]:
```suggestion
pip install poetry
```

File: docs/tutorials/developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+```
+pip install pipenv
+```
+Then install all dependencies including the ones for developers:
+```
+pipenv install -d
+pipenv shell
```
Comment (MrCloudSec) [inline docs/tutorials/developer-guide.md]:
```suggestion
poetry install
poetry shell
```

**Review by MrCloudSec — COMMENTED** [review]
Add poetry

**Review by jfagoagas — APPROVED** [review]
🔝

**Pull request merged**